beancount/parser/parser.c

```diff
@@ -223,7 +223,7 @@ static PyObject* parser_iternext(Parser* self)
     }
 
     /* Yield a (token name, line, matched string, token value) tuple. */
-    return Py_BuildValue("(sis#O)",
+    return Py_BuildValue("(siy#O)",
```

Comment:
Why change the iterator to return bytes?

Reply:
Because Flex `yyget_text()` returns a view into the lexer buffer, and those are bytes. This has been working as is so far because the lexer was not returning the matched text for string literals. In a the next commit this will change and thus this will break if the strings are not UTF-8 (and there are tests checking that other encodings are accepted). The easiest fix is to return byte objects here. Another possibility is to accept UTF-8 only, but you are worried about breaking compatibility there. Returning strings would require decoding the matched text for string literals and other token differently. Returning bytes is also (marginally) faster.